pkg/service/endpoints_controller.go

```diff
@@ -73,44 +73,48 @@ func (e *EndpointController) SyncServiceEndpoints() error {
 		for i := range pods.Items {
 			pod := &pods.Items[i]
 
-			// TODO: Once v1beta1 and v1beta2 are EOL'ed, this can
-			// assume that service.Spec.TargetPort is populated.
-			_ = v1beta1.Dependency
-			_ = v1beta2.Dependency
-			// TODO: Add multiple-ports to Service and expose them here.
-			portName := ""
-			portProto := service.Spec.Protocol
-			portNum, err := findPort(pod, service)
-			if err != nil {
-				glog.Errorf("Failed to find port for service %s/%s: %v", service.Namespace, service.Name, err)
-				continue
-			}
-			if len(pod.Status.PodIP) == 0 {
-				glog.Errorf("Failed to find an IP for pod %s/%s", pod.Namespace, pod.Name)
-				continue
-			}
+			for i := range service.Spec.Ports {
+				servicePort := &service.Spec.Ports[i]
+
+				// TODO: Once v1beta1 and v1beta2 are EOL'ed, this can
+				// assume that service.Spec.TargetPort is populated.
```

Comment:
A holdover from when we thought we were deprecating named ports, I think.
 removed

On Mon, Mar 30, 2015 at 1:57 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/service/endpoints_controller.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6182#discussion_r27431154>
> :
>
> > -			portName := ""
> > -			portProto := service.Spec.Protocol
> > -			portNum, err := findPort(pod, service)
> > -			if err != nil {
> > -				glog.Errorf("Failed to find port for service %s/%s: %v", service.Namespace, service.Name, err)
> > -				continue
> > -			}
> > -			if len(pod.Status.PodIP) == 0 {
> > -				glog.Errorf("Failed to find an IP for pod %s/%s", pod.Namespace, pod.Name)
> > -				continue
> > -			}
> > +			for i := range service.Spec.Ports {
> > +				servicePort := &service.Spec.Ports[i]
> > +
> > +				// TODO: Once v1beta1 and v1beta2 are EOL'ed, this can
> > +				// assume that service.Spec.TargetPort is populated.
>
> Isn't TargetPort set in defaulting?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6182/files#r27431154>
> .
>